packages/contracts-bedrock/src/L1/BatchVerifier.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ISemver} from "interfaces/universal/ISemver.sol";
+import {IEspressoNitroTEEVerifier} from "@espresso-tee-contracts/interface/IEspressoNitroTEEVerifier.sol";
+import {EspressoTEEVerifier} from "@espresso-tee-contracts/EspressoTEEVerifier.sol";
+import {IEspressoTEEVerifier} from "@espresso-tee-contracts/interface/IEspressoTEEVerifier.sol";
+
+interface INitroValidator {
+    function decodeAttestationTbs(
+        bytes memory attestation
+    )
+        external
+        pure
+        returns (bytes memory attestationTbs, bytes memory signature);
+}
+
+contract BatchVerifier is ISemver, OwnableUpgradeable {
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Mapping of batches verified by this contract
+    mapping(bytes32 => bool) public validBatches;
+
+    address public immutable preApprovedBatcher;
+
+    EspressoTEEVerifier public immutable espressoTEEVerifier;
+    INitroValidator public immutable nitroValidator;
+
+    constructor(
+        EspressoTEEVerifier _espressoTEEVerifier,
+        address _preApprovedBatcher
+    ) OwnableUpgradeable() {
+        espressoTEEVerifier = _espressoTEEVerifier;
+        preApprovedBatcher = _preApprovedBatcher;
+        nitroValidator = INitroValidator(
+            address(espressoTEEVerifier.espressoNitroTEEVerifier())
+        );
+    }
+
+    function decodeAttestationTbs(
+        bytes memory attestation
+    ) external view returns (bytes memory, bytes memory) {
+        return nitroValidator.decodeAttestationTbs(attestation);
+    }
+
+    function verifyBatch(
+        bytes32 commitment,
+        bytes calldata _signature
+    ) external {
+        // https://github.com/ethereum/go-ethereum/issues/19751#issuecomment-504900739
+        bytes memory signature = _signature;
+        uint8 v = uint8(signature[64]);
```

Comment:
Don't we have control on how the signature is computed? In this case can't we set the value $v$ to 27 or 28 right away?

Reply:
Ok I just read section `25.1.2 # Signature V Overview of the Book`. Sorry for the noise.